dev-docs/RFCs/proposals/color-filter-extension-rfc.md

```diff
@@ -0,0 +1,126 @@
+# RFC: Color Filter and Effects Extension for deck.gl Layers
+
+* Authors: Chris Gervang
+* Date: January, 2023
+* Status: **Draft**
+
+## Abstract
+
+This RFC proposes a new layer extension for `deck.gl` that allows users to easily apply color filters and effects, such as tint, brightness, and saturation, to any layer. The extension utilizes the `getModels` function from each layer and injects a `DECKGL_FILTER_COLOR` shader extension into each model.
+
+## Background
+
+`deck.gl` currently lacks a straightforward way to apply color filters and effects directly to layers. Users often have to write custom layers to achieve desired visual effects, which can be cumbersome. Alternatively, they are limited to using post-processing effects, which apply to the whole frame buffer rather than per layer.
+
+## Proposals
+
+### Design 1: User-Provided Luma ShaderPass
+
+#### Overview
+
+This design allows users to provide a `luma.gl` ShaderPass, such as the built-in vibrance module, to apply effects. The extension will utilize this ShaderPass for layer rendering.
+
+#### Implementation
+
+- Introduce a `ColorFilterExtension` that accepts a ShaderPass as an input parameter.
+- Modify the layer's fragment shader by integrating the provided ShaderPass at render time.
+
+```js
+import {
+  brightnessContrast,
+  hueSaturation,
+} from '@luma.gl/shadertools'
+
+new Tile3DLayer({
+    extensions: [new ColorFilterExtension(brightnessContrast)]
+    brightnessContrast: {
+        brightness: 0.5,
+        contrast: 0.5
+    }
```

Comment:
By importing the module you can easily obtain the props required using Typescript:

```ts
brightnessContrast: {
  brightness: 0.5,
  contrast: 0.5
} as typeof brightnessContrast.props
```